src/librustc/mir/interpret/allocation.rs

```diff
@@ -581,8 +581,10 @@ impl<'tcx, Tag, Extra> Allocation<Tag, Extra> {
 /// Run-length encoding of the undef mask.
 /// Used to copy parts of a mask multiple times to another allocation.
 pub struct AllocationDefinedness {
+    /// The lengths of ranges that are run-length encoded.
     ranges: smallvec::SmallVec::<[u64; 1]>,
-    first: bool,
+    /// The definedness of the first range.
+    initial: bool,
```

Comment:
Tiny nit (but I thought of it and now I can't not say it^^): might make more sense to make `initial` the first field.
When someone sees this type, I think it is easier to understand if they see this one first.

Then the doc comment for `ranges` can just say "at the end of each range, the boolean state flips", or something like that.

Reply:
I see the point, now the doc of `ranges` references `initial` and ready well in natural sequence order.